Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM registry.access.redhat.com/ubi9/go-toolset:1.20.10 AS builder
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+
+WORKDIR /qe-tools
+
+RUN git clone --branch main https://github.com/redhat-appstudio/qe-tools.git . && \
```

Comment:
I got this error when trying to build the image locally
```
Step 5/8 : RUN git clone --branch main https://github.com/redhat-appstudio/qe-tools.git . &&     CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o qe-tools main.go
 ---> Running in a670becb3a2d
Cloning into '.'...
/qe-tools/.git: Permission denied
The command '/bin/sh -c git clone --branch main https://github.com/redhat-appstudio/qe-tools.git . &&     CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o qe-tools main.go' returned a non-zero code: 1
```

I think instead of cloning the repo inside Dockefile we could just COPY files from the system into builder, because if we're going to run docker-build workflow per PR, we need to validate PR changes